lib/node_modules/@stdlib/array/base/mskfilter-map/docs/types/test.ts

```diff
@@ -0,0 +1,271 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import mskfilterMap = require( './index' );
+
+/**
+* Unary function.
+*
+* @param value - input value
+* @returns result
+*/
+function clbk( value: number ): number {
+	return value;
+}
+
+// TESTS //
+
+// The function returns a nested array...
+{
+	const x = [ 1.0, 2.0, 3.0, 4.0 ];
+
+	mskfilterMap( x, [ 1, 1, 1, 1 ], clbk ); // $ExpectType number[]
+	mskfilterMap( x, [ 1, 1, 1, 1 ], clbk, {} ); //  $ExpectType number[]
+}
+
+// The function returns an array...
+{
+	mskfilterMap( [ 1, 2, 3, 4 ], [ 1, 1, 1, 1 ], clbk ); // $ExpectType number[]
+	mskfilterMap( [ 1, 2, 3, 4 ], [ 1, 1, 1, 1 ], clbk, {} ); // $ExpectType number[]
+}
+
+// The compiler throws an error if the function is provided a first argument which is not an array-like object...
+{
+	mskfilterMap( 1, [ 0, 0 ], clbk ); // $ExpectError
+	mskfilterMap( true, [ 0, 0 ], clbk ); // $ExpectError
+	mskfilterMap( false, [ 0, 0 ], clbk ); // $ExpectError
+	mskfilterMap( null, [ 0, 0 ], clbk ); // $ExpectError
+	mskfilterMap( void 0, [ 0, 0 ], clbk ); // $ExpectError
+	mskfilterMap( {}, [ 0, 0 ], clbk ); // $ExpectError
+
+	mskfilterMap( 1, [ 0, 0 ], clbk, {} ); // $ExpectError
+	mskfilterMap( true, [ 0, 0 ], clbk, {} ); // $ExpectError
+	mskfilterMap( false, [ 0, 0 ], clbk, {} ); // $ExpectError
+	mskfilterMap( null, [ 0, 0 ], clbk, {} ); // $ExpectError
+	mskfilterMap( void 0, [ 0, 0 ], clbk, {} ); // $ExpectError
+	mskfilterMap( {}, [ 0, 0 ], clbk, {} ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided a second argument which is not an array-like object containing numbers...
```

Comment:
```suggestion
// The compiler throws an error if the function is provided a second argument which is not an array-like object...
```

Currently, we don't require that `mask` only contain numbers. At least that is not what you specify in `index.d.ts`.